shared/iputils/external_ip.go

```diff
@@ -7,10 +7,64 @@ import (
 
 // ExternalIPv4 returns the first IPv4 available.
 func ExternalIPv4() (string, error) {
-	ifaces, err := net.Interfaces()
+	ips, err := retrieveIPAddrs()
+	if err != nil {
+		return "", err
+	}
+	if len(ips) == 0 {
+		return "127.0.0.1", nil
+	}
+	for _, ip := range ips {
+		ip = ip.To4()
+		if ip == nil {
+			continue // not an ipv4 address
+		}
+		return ip.String(), nil
+	}
+	return "127.0.0.1", nil
+}
+
+// ExternalIPv6 retrieves any allocated IPv6 addresses
+// from the accessible network interfaces.
+func ExternalIPv6() (string, error) {
```

Comment:
I realized `ExternalIPv6() ` is not covered in `external_ip_test.go `. Should we cover it?

Reply:
The issue would be in bazels sandbox the ipv6 address isnt availble from the accessible network interface which makes it hard to test something like this.